addons/mail/controllers/webclient.py

```diff
@@ -85,6 +85,10 @@ def _process_request_for_logged_in_user(self, store: Store, name, params):
             # sudo: mail.notification - return only failures of current user as author
             notifications = request.env["mail.notification"].sudo().search(domain, limit=100)
             notifications.mail_message_id._message_notifications_to_store(store)
+        if name == "avatar_card":
+            user_id = params["user_id"]
+            user = request.env["res.users"].browse(user_id)
+            self._handle_avatar_card_request(store, user)
```

Comment:
Should use search to ensure the user exists and acl are correct (and maybe active_test false, at least for odoobot), and do nothing if not found

```suggestion
            if user := request.env["res.users"].with_context(active_test=False).search([("id", "=", params["user_id"])]):
                self._handle_avatar_card_request(store, user)
```